app/commands/assigning.rb

```diff
@@ -23,38 +25,112 @@ def stream_name
   class AssignToUser < Assigning::Command
     attribute :user_id, Types::Uuid
     attribute :user_name, Types::String
+    attribute :to_whom_id, Types::Uuid
+    attribute :to_whom_name, Types::String
+
+    def call
+      publish(AssignedToUser, data: to_hash)
+    end
+  end
+
+  class AssignToSelf < Assigning::Command
+    attribute :user, Types.Instance(User)
+
+    def call
+      AssignToUser.new(
+        crime_application_id: crime_application_id,
+        user_id: user.id,
+        user_name: user.name,
+        to_whom_id: user.id,
+        to_whom_name: user.name
+      ).call
+    end
+  end
+
+  class ReassignToSelf < Assigning::Command
+    attribute :user, Types.Instance(User)
+    attribute :state_key, Types::String
+
+    def call
+      validate_state
+
+      ReassignToUser.new(
+        crime_application_id: crime_application_id,
+        user_id: user.id,
+        user_name: user.name,
+        from_whom_id: current_assignment.user_id,
+        from_whom_name: current_assignment.user_name,
+        to_whom_id: user.id,
+        to_whom_name: user.name
+      ).call
+    end
+
+    private
+
+    def current_assignment
+      @current_assignment ||= CurrentAssignment.new(
+        crime_application_id:
+      )
+    end
+
+    def validate_state
+      raise StateHasChanged if state_changed?
+    end
+
+    def state_changed?
+      state_key != current_assignment.state_key
+    end
+  end
+
+  class ReassignToUser < Assigning::Command
```

Comment:
Are reassignments the same thing as assignments? We have seperate controllers wondering if they should be seperately namespaced